lib/api/errors/validation.rb

```diff
@@ -53,19 +53,21 @@ def self.create(errors)
 
       # Merges property error messages (e.g. for status and status_id)
       def self.merge_error_properties(errors)
-        properties = errors.keys
+        merged_errors = errors.dup
 
-        properties.each do |p|
-          match = /(?<property>\w+)_id/.match(p)
+        errors.keys.each do |property|
+          match = /(?<property>\w+)_id/.match(property)
 
           if match
             key = match[:property].to_sym
-            error = Array(errors[key]) + errors[p]
+            error = Array(errors[key]) + errors[property]
 
-            errors.set(key, error)
-            errors.delete(p)
+            merged_errors.set(key, error)
+            merged_errors.delete(property)
```

Comment:
Hm, somebody forgot to write specs for the whole class. As you fixed an existing bug, I think you should at least write specs for `merge_error_properties`.